source/hash_table.c

```diff
@@ -435,14 +454,23 @@ static struct hash_table_entry *s_emplace_item(
         }
     }
 
-    return initial_placement;
+    AWS_POSTCONDITION(hash_table_state_is_valid(state));
+    AWS_POSTCONDITION(rval >= &state->slots[0] && rval < &state->slots[state->size]);
+    return rval;
 }
 
 static int s_expand_table(struct aws_hash_table *map) {
     struct hash_table_state *old_state = map->p_impl;
     struct hash_table_state template = *old_state;
 
-    s_update_template_size(&template, template.size * 2);
+    size_t new_size;
+    if (aws_mul_size_checked(template.size, 2, &new_size)) {
+        return AWS_OP_ERR;
+    }
+
+    if (s_update_template_size(&template, new_size)) {
```

Comment:
Is it really okay to change the behaviour of the code here? Why wasn't the return code being checked before?

Reply:
The return code wasn't being checked because it was a bug.